src/main/java/io/numaproj/kafka/KafkaApplication.java

```diff
@@ -1,20 +1,39 @@
 package io.numaproj.kafka;
 
-import io.numaproj.numaflow.sinker.Server;
+import io.numaproj.kafka.consumer.Admin;
+import io.numaproj.kafka.consumer.KafkaSourcer;
+import io.numaproj.kafka.consumer.Worker;
+import io.numaproj.kafka.producer.KafkaSinker;
+import io.numaproj.kafka.schema.ConfluentRegistry;
+import io.numaproj.kafka.schema.Registry;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
 
 @Slf4j
 @SpringBootApplication
+@EnableConfigurationProperties
+@ComponentScan(
```

Comment:
Might be better off to create a separate folder for all the source and producer related code.   Then create a separate configuration file where you can add the package you want to scan depending on source or sink.